csrc/codegen.cpp

```diff
@@ -3424,6 +3440,21 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     // Something like:
     //   myFunc(T1[0], T0[0]);
     if (as_utility) {
+      if (!asm_->options().immediate_inputs.empty()) {
+        code_ << "<";
+        bool first = true;
+        for (auto [constraint, register_] : asm_->constraintsAndInputs()) {
```

Comment:
Use rvalue reference for `std::string`
```suggestion
        for (auto&& [constraint, register_] : asm_->constraintsAndInputs()) {
```

Reply:
changed